### PR TITLE
Add CrashlyticsReport persistence

### PR DESCRIPTION
**Pull request by mrwillis21**
Serializes to JSON files on disk

### CODE REVIEW

File: firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistenceTest.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+                .setProximityOn(true)
+                .setRamUsed(10000000)
+                .build())
+        .build();
+  }
+
+  private static ImmutableList<Frame> makeTestFrames() {
+    return ImmutableList.from(
+        Frame.builder()
+            .setPc(0)
```
Comment (bklimt) [inline firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistenceTest.java]: It's probably a better test if these aren't all 0.

File: firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistence.java [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+      while ((read = fileInput.read(readBuffer)) > 0) {
+        bos.write(readBuffer, 0, read);
+      }
+      return new String(bos.toByteArray(), UTF_8);
+    } catch (IOException e) {
+      return null;
+    }
+  }
+
+  private static void recursiveDelete(File file) {
```
Comment (bklimt) [inline firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistence.java]: These methods already deleted elsewhere in the SDK, correct? So is the plan to delete the other versions?

Comment (mrwillis21) [reply firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistence.java]: Ultimately, yes. My goal in the long term is to reduce "Utils" classes, so for now I went with the Go proverb "A little copying is better than a little dependency."

File: firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistenceTest.java (hunk 1 above)
Comment (mrwillis21) [reply firebase-crashlytics/src/androidTest/java/com/google/firebase/crashlytics/internal/persistence/CrashlyticsReportPersistenceTest.java]: Good call, I'll update this.

**Pull request merged**